File: ios/Samples/Showcase-ObjectiveC/Showcase-ObjectiveC/ViewControllers/Files/VideoDisplayViewController.m
//
//  AudioDisplayViewController.m
//  LiferayScreens-Showcase-ObjectiveC
//
//  Created by Victor Galán on 04/02/2017.
//  Copyright © 2017 liferay. All rights


@import LiferayScreens;
#import "LiferayLogger.h"
#import "VideoDisplayViewController.h"

@interface VideoDisplayViewController () <FileDisplayScreenletDelegate>

@property (weak, nonatomic) IBOutlet VideoDisplayScreenlet *screenlet;

@end

@implementation VideoDisplayViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    self.screenlet.delegate = self;
    self.screenlet.presentingViewController = self;
    self.screenlet.classPK = [LiferayServerContext longPropertyForKey:@"videoDisplayClassPK"];
}

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetError:(NSError *)error {
    LiferayLog(error);
}

- (void)screenlet:(FileDisplayScreenlet *)screenlet onFileAssetResponse:(NSURL *)url {
    LiferayLog(url);
}

@end

File: xamarin/Samples/Showcase-iOS/ViewController/VideoDisplayViewController.cs
using Foundation;
using LiferayScreens;
using System;
using System.Diagnostics;
using UIKit;

namespace ShowcaseiOS.ViewController
{
    public partial class VideoDisplayViewController : UIViewController, IFileDisplayScreenletDelegate
    {
        public VideoDisplayViewController(IntPtr handle) : base(handle) { }

        public override void ViewDidLoad()
        {
            base.ViewDidLoad();

            this.videoDisplayScreenlet.ClassPK = 38943;
            this.videoDisplayScreenlet.ClassName = "com.liferay.document.library.kernel.model.DLFileEntry";

            this.videoDisplayScreenlet.Delegate = this;
        }

        /* IFileDisplayScreenletDelegate */

        [Export("screenlet:onFileAssetError:")]
        public virtual void OnFileAssetError(FileDisplayScreenlet screenlet, NSError error)
        {
            Debug.WriteLine($"Video display failed: {error.DebugDescription}");
        }

        [Export("screenlet:onFileAssetResponse:")]
        public virtual void OnFileAssetResponse(FileDisplayScreenlet screenlet, NSUrl url)
        {
            Debug.WriteLine($"Video display success: {url.DebugDescription}");
        }
    }
}

